### ISSUE TITLE
Custom tickets: record change log

### ISSUE THREAD

**Issue by carmenfan** [3drepo/3drepo.io#3578]
### Description
Everytime there is a change on the ticket, we want to add a record to log it.

For e.g. if the following update is applied on a ticket:
````js
{
  properties: {
    xyz: "new val"
  }
}
````

Then we want to store a record in `tickets.logs` that shows:
````
{
  _id: "generate an id", //uuid
  teamspace: "teamspace",
  project: "project", //note this should be UUID not string
  model: "model",
  ticket: "ticket id" , //note this should be UUID not string
  author: "username", // the user who made the change
  from: {
    properties: { xyz: "old val" }
  },
  to: { 
    properties: {xyz: "new val"}
  },
  date: new Date(), //date at which this was updated, this should be identifical to `properties.Updated at` at that point in time.
}
````

#### NOTES:
- notice no model ID on the collection - we're moving towards storing all tickets from all con/fed in the same collection.
- when we remove a model, we need to remove the ticket data from the collection
- it's possible that there are multiple properties/module properties being updated
- No need to add a log for new ticket - only updates

### Goals
- [ ] Add a system log every time a ticket has been updated

### Tasks
To be filled in by the developer

### Related Resources
Will depend on #3564
